lambdas/handlers/authoriser_handler.py

```diff
@@ -0,0 +1,268 @@
+"""
+This code has been modified from AWS blueprint. Below is the original license:
+
+Copyright 2015-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License. A copy of the License is located at
+     http://aws.amazon.com/apache2.0/
+or in the "license" file accompanying this file.
+This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and limitations under the License.
+"""
+
+import logging
+import os
+import re
+import time
+
+import boto3
+import botocore.exceptions
+import jwt
+from boto3.dynamodb.conditions import Key
+from enums.permitted_role import PermittedRole
+from services.dynamo_service import DynamoDBService
+from utils.exceptions import AuthorisationException
+
+from utils.get_aws_region import get_aws_region
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def lambda_handler(event, context):
+    user_roles = []
+    ssm_public_key_parameter_name = os.environ["SSM_PARAM_JWT_TOKEN_PUBLIC_KEY"]
+
+    try:
+        client = boto3.client("ssm", region_name=get_aws_region())
+        ssm_response = client.get_parameter(
+            Name=ssm_public_key_parameter_name, WithDecryption=True
+        )
+        public_key = ssm_response["Parameter"]["Value"]
+
+        decoded = jwt.decode(
+            event["authorizationToken"], public_key, algorithms=["RS256"]
+        )
+
+        ndr_session_id = decoded["ndr_session_id"]
+
+        current_session = find_login_session(ndr_session_id)
+        validate_login_session(current_session, ndr_session_id)
+
+        # if user has a valid session, assign their role
+        user_roles = [org["role"] for org in decoded["organisations"]]
+
+    except AuthorisationException as e:
+        logger.error(e)
+    except botocore.exceptions.ClientError as e:
+        logger.error(e)
+    except jwt.PyJWTError as e:
+        logger.error(f"error while decoding JWT: {e}")
+    except (KeyError, IndexError) as e:
+        logger.error(e)
```

Comment:
What's happening in these except blocks? Once the error is logged the code will continue on and return the auth response. Is this what we want?

Reply:
At the auth response part, if the user doesn't have a valid role, we responds with Deny All by default.
I have edited these except block and immediately respond with a Deny All here